integration/integration.go

```diff
@@ -3,41 +3,110 @@ package integration
 import (
 	"bytes"
 	"encoding/json"
+	"fmt"
 	"io"
+	"os"
+	"reflect"
 	"sync"
 
+	"github.com/newrelic/infra-integrations-sdk/args"
+	"github.com/newrelic/infra-integrations-sdk/log"
 	"github.com/newrelic/infra-integrations-sdk/persist"
 	"github.com/pkg/errors"
 )
 
+const protocolVersion = "2"
+
 // Integration defines the format of the output JSON that integrations will return for protocol 2.
 type Integration struct {
-	locker             sync.Locker
-	storer             persist.Storer
 	Name               string    `json:"name"`
 	ProtocolVersion    string    `json:"protocol_version"`
 	IntegrationVersion string    `json:"integration_version"`
-	Data               []*Entity `json:"data"`
+	Entities           []*Entity `json:"data"`
+	locker             sync.Locker
+	storer             persist.Storer
 	prettyOutput       bool
 	writer             io.Writer
+	logger             log.Logger
+	args               interface{}
 }
 
 // New creates new integration with sane default values.
-func New(name, version string) (*Integration, error) {
-	return NewBuilder(name, version).Build()
+func New(name, version string, opts ...Option) (i *Integration, err error) {
+
+	if name == "" {
+		err = errors.New("integration name cannot be empty")
+		return
+	}
+
+	if version == "" {
+		err = errors.New("integration version cannot be empty")
+		return
+	}
+
+	i = &Integration{
+		Name:               name,
+		ProtocolVersion:    protocolVersion,
+		IntegrationVersion: version,
+		Entities:           []*Entity{},
+		writer:             os.Stdout,
+		locker:             disabledLocker{},
+		logger:             log.NewStdErr(false),
+	}
+
+	for _, opt := range opts {
+		if err = opt(i); err != nil {
+			err = fmt.Errorf("error applying option to integration. %s", err)
+			return
+		}
+	}
+
+	// arguments
+	if err = i.checkArguments(); err != nil {
+		return
+	}
+	if err = args.SetupArgs(i.args); err != nil {
```

Comment:
Now that I stop here at this line... Should not this be part of an option itself? The Setup + assignment to `i.prettyOutput` ?

Reply:
I guess so, but arguments will be changed in the near future